### DISCUSSION TITLE
[Bug] When choosing `Saved Replies` during PR approval, the UI locks up

### DISCUSSION THREAD

**Discussion by jazeee** [community/community#102662] [Pull Requests]
### Select Topic Area

Bug

### Body

Saved replies UI appears to be broken when using other modal dialog UIs

Steps to reproduce:
1. Open a previously created PR
2. Click on `Review Changes`
3. Observe the expected dialog titled `Finish your review`
4. Click on the `Saved Replies` icon.
Observe the unexpected - the UI is blocked from interactivity, and there is a hidden blocking UI element.
Expect to see a `Saved Replies` UI to pick a saved reply.

Note - this appears to only affect dialogs on top of dialogs.
Saved Replies still works in other parts of github UI.

https://www.loom.com/share/7001824b8a6a4d529f7f248db2a2c085

**Comment by obryckim** [comment]
I am seeing this as well. This has been happening for most of the day today for me.
If I go to my profile, I can edit/view my saved replies; however, in the PR review, I get the behavior described in this issue.

**Comment by JosephTLyons** [comment]
For me, when selecting a saved reply in a GitHub Issue, only the first line of the reply is inserted into the issue. This bug is not there for me in GitHub Discussions. This broke recently.

**Comment by shawnmartin-py** [comment]
**🔨 issue:** ‏I can also add them here, but not on `Review Changes`

**Comment by smcvb** [comment]
I use this feature frequently due to the amount of reviews I do through GitHub, but I can confirm the UI freezes the moment I select the `Saved Replies` button in the `Review changes` window.
A fix for this would be much appreciated :sweat_smile:

**Comment by antonjoel82** [comment]
Having similar issues. Context below:

While reviewing a PR, I have observed issues with using the "Select a Reply" functionality via <kbd>ctrl + .</kbd> shortcut. They were not present a few months ago. "Select a reply" is now a modal with background blur, but in the past was handled more as a popover IIRC.

Primary steps:

1. Add custom replies in your preferences
2. Click the `+` icon on a line of code during PR review
3. Type shortcut: <kbd>ctrl + .</kbd> to bring up "Select a reply"
4. Attempt to select a reply by typing text (e.g.  "Thought")
5. Type <kbd>Enter</kbd>

The issues are:
- During the step of typing to filter replies, the top-level GitHub / repo search comes into focus, so typing does not filter replies.
- When filtering replies, the up / down arrow keys no longer work to quickly highlight and select a filtered reply (and instead change cursor position in the input.
- Upon selecting a reply, focus does not return to the comment input.
- Occasionally, when using the keyboard shortcut and typing sequence, all ability to control the page (scrolling, clicking, etc) disappears, and I **must refresh the page** for it to work.

**Reply by jazeee** [reply]
One thing I noticed is that you can regain control of the UI by hitting the escape key

**Comment by jazeee** [comment]
As of 2024-02-02, it looks like we have partial restoration, however the replies dialog shows up behind the approval dialog

**Comment by jazeee** [comment]
This issue appears to be fully resolved.
